src/robot/libraries/Collections.py

```diff
@@ -387,7 +387,12 @@ def lists_should_be_equal(self, list1, list2, msg=None, values=True,
         default = 'Lengths are different: %d != %d' % (len1, len2)
         _verify_condition(len1 == len2, default, msg, values)
         names = self._get_list_index_name_mapping(names, len1)
-        diffs = list(self._yield_list_diffs(list1, list2, names))
+        if not ignore_order:
+            diffs = list(_yield_list_diffs(list1, list2, names))
```

Comment:
Some automatic checks are failing. I suspect, it is because of `self.` missed before `_yield_list_diffs` call.

Reply:
my Bad! Correcting it.

Reply:
Corrected and pushed code. Thanks @tysonite and @pekkaklarck  for quick review.